libraries/standard.js

```diff
@@ -81,6 +81,52 @@ const remove_if_empty = function (object, field) {
   }
 }
 
+
+const create_dates_template = function() {
+  return {
+    created: {
+      day: 0,
+      month: 0,
+      year: 0,
+      date_string: "",
+    },
+    installed: {
+        day: 0,
+        month: 0,
+        year: 0,
+        date_string: "",
+    },
+    commissioned: {
+        day: 0,
+        month: 0,
+        year: 0,
+        date_string: "",
+    }
+  }
+}
+
+const remove_null_date_fields = function(item) {
+  for (const property in item.dates) {
+    remove_if_zero(item.dates[property]);
+  }
+  for (const att in item.dates) {
+      if (Object.keys(item.dates[att]).length == 0) {
+          delete item.dates[att];
+      }
+  }
+  if (Object.keys(item.dates).length == 0) {
+      delete item.dates;
+  }
+  return item;
+}
+
+const remove_if_zero = function(object) {
+  for (const prop in object) {
+      if (object[prop] == 0) delete object[prop];
+      else if (object[prop] == "") delete object[prop];
```

Comment:
should this check for null and undefined too? if so, you can do:
else if (!object[prop]) delete object[props];

Reply:
I think 0 in this case is different than undefined or null. If we want to remove undefined/null values we can use remove_if_null but my assumption is that this is intended to work with the dates object which sets fields to 0 or "" by default.

Reply:
plus we don't export it in the std_lib map so I think its safe to leave as is